powersimdata/network/constants/plants.py

```diff
@@ -1,101 +1,439 @@
+    """
+
+    def __init__(self, model):
+        if model in ["usa_tamu", "hifld"]:
+            self.type2label = {
+                "nuclear": "Nuclear",
+                "geothermal": "Geo-thermal",
+                "coal": "Coal",
+                "dfo": "DFO",
+                "hydro": "Hydro",
+                "ng": "Natural Gas",
+                "solar": "Solar",
+                "wind": "Wind",
+                "wind_offshore": "Wind Offshore",
+                "biomass": "Biomass",
+                "other": "Other",
+                "storage": "Storage",
+            }
+            self.curtailable2label = {
+                "solar": "Solar Curtailment",
+                "wind": "Wind Curtailment",
+                "wind_offshore": "Offshore Wind Curtailment",
+            }
+        elif model == "europe_tub":
+            self.type2label = {
+                "onwind": "Onshore Wind",
+                "offwind-ac": "Offshore Wind (AC)",
+                "offwind-dc": "Offshore Wind (DC)",
+                "hydro": "Reservoir & Dam",
+                "PHS": "Pumped Hydro Storage",
+                "ror": "Run of River",
+                "solar": "Solar",
+                "biomass": "Biomass",
+                "geothermal": "Geo-thermal",
+                "OCGT": "Open-Cycle Gas",
+                "CCGT": "Combined-Cycle Gas",
+                "nuclear": "Nuclear",
+                "coal": "Coal",
+                "lignite": "Lignite",
+                "oil": "Oil",
+                "H2": "Hydrogen Storage",
+                "battery": "Battery Storage",
+            }
+            self.curtailable2label = {
+                "solar": "Solar Curtailment",
+                "onwind": "Onshore Wind Curtailment",
+                "offwind-ac": "Offshore Wind Curtailment (AC)",
+                "offwind-dc": "Offshore Wind Curtailment (DC)",
+            }
+        self.label2type = {v: k for k, v in self.type2label.items()}
+        self.label2curtailable = {v: k for k, v in self.curtailable2label.items()}
+
+
+class Efficiency:
+    """Efficiency for thermal generators.
+
+    For US model:
+
+    * Mwh electric to MWh thermal: `Danish Energy Agency, Technology Data - Generation
+    of Energy and District Heating <https://ens.dk/sites/ens.dk/files/Analyser/technology_data_catalogue_for_el_and_dh.pdf>`_.
+
+    :param str model: grid model.
+    """
+
+    def __init__(self, model):
+        if model in ["usa_tamu", "hifld"]:
+            self.efficiency = {
+                "coal": 0.33,
+                "dfo": 0.35,
+                "ng": 0.41,  # referring to OCGT values from Danish Energy Agency
+            }
+        elif model == "europe_tub":
+            self.efficiency = {
+                "biomass": 0.416,
+                "OCGT": 0.39,
+                "CCGT": 0.55,
+                "coal": 0.33,
+                "lignite": 0.33,
+                "oil": 0.35,
+            }
+
+
+class Emission:
```

Comment:
Tempting to split up the classes such that we remove the if/else logic based on the grid model. Currently this would mainly be for readability, but could also reduce the effort to extend this to other grids. E.g. a user would only need to define a new class, and modify one line of the existing code (ideally it would be extensible without any modification of existing code). 

Example of what this could look like: define new `USAEmission` and `EuropeEmission` classes that only set the values per grid, e.g. `carbon_per_mwh`, `so2_per_mwh`, etc. Also add a function 
```
def get_emission(model):
    _lookup = {
        "usa_tamu": USAEmission,
        "hifld": USAEmission,
        "europe_tub": EuropeEmission,
    }
    return _lookup[model]().__dict__

```
Then rather than calling `Emission(model).__dict__` within `get_plants` we would have `get_emission(model)`. 

Lemme know what you think, and feel free to save for a follow up PR if we want to use this approach, since it requires changing a lot in this file.

Reply:
I like the idea of removing `if/else` logic in these look up tables.

Reply:
Sounds good, I will work on it

Reply:
Done in d34d8bd